### PR TITLE
Add more functional testing for SQLite

### PR DESCRIPTION
**Pull request by natemcmaster**
Exercise more of that SQLite provider :muscle:. Only adds code to testing.

### CODE REVIEW

**Comment by natemcmaster** [comment]
NB: I will squash to one commit before merging.

File: test/EntityFramework.SqlServer.FunctionalTests/MappingQuerySqlServerFixture.cs [hunk 1]
```diff
@@ -43,6 +43,8 @@ public void Dispose()
             _testDatabase.Dispose();
         }
 
+        protected override string DatabaseSchema { get; } = "dbo";
```
Comment (bricelam) [inline test/EntityFramework.SqlServer.FunctionalTests/MappingQuerySqlServerFixture.cs]: Ultimately, Sqlite will ignore schemas and throw if you try to configre them specifically for Sqlite. (Part of #1132)

Comment (natemcmaster) [inline test/EntityFramework.SqlServer.FunctionalTests/MappingQuerySqlServerFixture.cs]: It should, but Sqlite is still including schema name in queries. This was required to get MappingQuery tests to pass for Sqlite 

```
Test 'MappingQuerySqliteTest.All_orders' failed: Microsoft.Data.Sqlite.SqliteException : 
      Error: no such table: dbo.Orders
```

File: test/EntityFramework.Sqlite.FunctionalTests/SqliteTestStore.cs [hunk 2]
```diff
@@ -53,6 +54,34 @@ private SqliteTestStore CreateTransient()
...
+        {
+            var command = _connection.CreateCommand();
+
+            if (_transaction != null)
+            {
+                command.Transaction = _transaction;
+            }
+
+            command.CommandText = commandText;
+            command.CommandTimeout = CommandTimeout;
```
Comment (bricelam) [inline test/EntityFramework.Sqlite.FunctionalTests/SqliteTestStore.cs]: This doesn't do anything in the ADO.NET provider.

File: test/EntityFramework.Sqlite.FunctionalTests/SqliteTestStore.cs [hunk 3]
```diff
@@ -53,6 +54,34 @@ private SqliteTestStore CreateTransient()
...
+
+            for (var i = 0; i < parameters.Length; i++)
+            {
+                command.Parameters.AddWithValue("p" + i, parameters[i]);
```
Comment (bricelam) [inline test/EntityFramework.Sqlite.FunctionalTests/SqliteTestStore.cs]: Should this include the '@' symbol?

File: test/EntityFramework.SqlServer.FunctionalTests/MappingQuerySqlServerFixture.cs (hunk 1 above)
Comment (bricelam) [inline test/EntityFramework.SqlServer.FunctionalTests/MappingQuerySqlServerFixture.cs]: Fine for now. I'll update it when I break it.

**Comment by bricelam** [comment]
:shipit:

**Comment by bricelam** [comment]
Were the ones not included in this PR (GraphUpdatesTest, IncludeOneToOneTest, MonsterFixupTest & TransactionTest) the flaky ones?

**Comment by natemcmaster** [comment]
Yes, I didn't include these tests in this PR because there are several dozen failing. I'm working on those right now to identify the cause of the failures.

File: test/EntityFramework.Sqlite.FunctionalTests/SqliteTestStore.cs (hunk 3 above)
Comment (natemcmaster) [inline test/EntityFramework.Sqlite.FunctionalTests/SqliteTestStore.cs]: Good catch. Yes, this should use '@'

**Pull request merged**